content/extending-ui/extending-chat-widget/rich-messages/index.mdx

```diff
@@ -0,0 +1,272 @@
+
+![Carousel Placeholder](images/carousel-placeholder.png)
+
+For example, by just showing a title and image, you can present multiple products in a more appealing way.
+
+## Quick replies
+Quick replies are different types of buttons - they are smaller, so that much more of them can fit on the screen:
+
+![Quick Replies Placeholder](images/quick-replies-placeholder.png)
+
+They might be useful when you want your users to choose one of the predefined responses.
+
+# Getting started
+As mentioned, Rich Messages are combination of basic building blocks. How do we know what to show then? We base that on the JSON object received through the [Customer Chat API](../../../messaging/customer-chat-api/#rich-message). Apart from the author and other, strictly related to messaging data, the core of the Rich Messages looks like that:
+
+<CodeSample>
+
+```json
+{
+	"template_id": "cards",
+	"elements": [{
+            "title": "Title",
+			"subtitle": "Subtitle",
+			"image": {
+                "name": "example.png",
+				"url": "https://example.org/image.png",
+                "content_type": "image/png",
+                "size": 1024,
+                "width": 640,
+                "height": 480
+			},
+			"buttons": [{
+				"type": "message",
+				"text": "text",
+				"postback_id": "send_message",
+				"user_ids": [],
+				"value": "text"
+			}]
+		}
+	]
+}
+```
+
+</CodeSample>
+
+This is how it would look like in the Chat Widget:
+
+![Getting Started Placeholder](images/getting-started-placeholder.png)
+
+Let's go through each of the elements:
+
+* `template_id` - defines how the Rich Message would be presented. Could be either `cards`, which covers both Cards and Carousels, or `quick_replies`, which corresponds to Quick Replies. This is a required parameter.
+* `elements` - optional, this is the array of elements included in the Rich Message, for example multiple cards. There can be a maximum of 11 elements.
+* `elements.title` and `elements.subtitle` - both optional, displays the formatted text on Cards.
+* `elements.image` - optional, displays the image on the card. Can have the following optional parameters: `name`, `url`, `content_type`, `size`, `width`, `height`. 
+* `elements.buttons` - optional, array of the buttons displayed (maximum of 11). They can have the following parameters (all required if the `elements.buttons` is included):
+    * `elements.buttons.type` - defines the behavior after user clicks on the button. Might have the following values:
+        * `webview` - opens the [Moment](/extending-ui/extending-chat-widget/chat-widget-moments/),
+        * `message` - sends a message postback in the chat,
+        * `url` - opens given URL in the new tab,
+        * `phone` - allows the Customer to call the number given directly from the chat,
+    * `elements.buttons.postback_id` - description of the action sent,
+    * `elements.buttons.user_ids` - describes the user that sent the postback, can be empty,
```

Comment:
user_ids array needs clarification - it's an array of users who clicked this button